projects/libyuv/libyuv_fuzzer.cc

```diff
@@ -0,0 +1,80 @@
+/* # Copyright 2020 Google Inc. */
+/* # */
+/* # Licensed under the Apache License, Version 2.0 (the "License"); */
+/* # you may not use this file except in compliance with the License. */
+/* # You may obtain a copy of the License at */
+/* # */
+/* #      http://www.apache.org/licenses/LICENSE-2.0 */
+/* # */
+/* # Unless required by applicable law or agreed to in writing, software */
+/* # distributed under the License is distributed on an "AS IS" BASIS, */
+/* # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. */
+/* # See the License for the specific language governing permissions and */
+/* # limitations under the License. */
+/* # */
+/* ################################################################################ */
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include "libyuv.h"
+#include "fuzz_common.h"
+
+using namespace std;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+
+    FuzzedDataProvider fuzzed_data(data, size);
```

Comment:
This doesn't allocate enough memory for the source.
The amount of bytes consumed is
   int range_size = sqrt((static_cast<int>(size) + 1) / 2);
    size_t width = fuzzed_data.ConsumeIntegralInRange(1, range_size);
    size_t height = fuzzed_data.ConsumeIntegralInRange(1, range_size);
    int kPixels = width * height;
    int kHalfPixels = ((width + 1) / 2) * ((height + 1) / 2);
    size_t  new_size = kPixels + kHalfPixels * 2

libyuv::I420ToNV21 will consume new_size bytes if you pass these width and height parameters.

For example if you were passed a size of 1, your code computes width and height of 1.
the Y plane is 1x1 so 1 byte
the U and V planes are half width and height but round up to 1x1 so 1 byte each
total of 3 bytes consumed.

theres really no valid 1 byte YUV image.  its 3 planes and each plane is a minimum of 1 byte.
if you're working backward from bytes, you'll need to be careful about how you compute width and height if you need to stay under 'size' bytes.
Your width and height assume 1 plane I think"?
Come up with a formula where width * height + ((width + 1) / 2) * ((height + 1) / 2) * 2 is <= size
e.g. for 4 bytes, width and height are still 1.  or 2x1.  but not 2x2.  2x2 takes 6 bytes.
it doesnt need to be square and for good code coverage it probably shouldnt be.
for example and interesting case is 1x16
a width of 1 requires the rows of u and v to be rounded out to 1
y = 1x16 = 16 bytes
u = 1x8
v = 1x8
total is 32 bytes.

Reply:
I modify my code and show blew.
>int range_size = (static_cast<int>(size) + 1) / 2;
   do
   {
         height = fuzzed_data.ConsumeIntegralInRange(0, range_size);
         width = fuzzed_data.ConsumeIntegralInRange(0, range_size) ;
         kPixels = width * height;
         kHalfPixels = (width + 1) / 2 * (height + 1) / 2;
    }while (kPixels + kHalfPixels * 2 > size);
`